csharp/extractor/Semmle.Extraction.CSharp/Entities/PreprocessorDirectives/LineOrSpanDirective.cs

```diff
@@ -0,0 +1,32 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.IO;
+
+namespace Semmle.Extraction.CSharp.Entities
+{
+    internal enum LineDirectiveKind
+    {
+        Default = 0,
+        Hidden = 1,
+        Numeric = 2,
+        Span = 3
+    }
+
+    internal abstract class LineOrSpanDirective<T> : PreprocessorDirective<T> where T : LineOrSpanDirectiveTriviaSyntax
+    {
+        protected LineOrSpanDirective(Context cx, T trivia)
+            : base(cx, trivia)
+        {
+        }
+
+        protected override void PopulatePreprocessor(TextWriter trapFile)
```

Comment:
I think I would add a `LindeDirectiveKind kind` field to this class, which should be passed in to the constructor, and then do `trapFile.directive_lines(this, kind);` here instead of in the base classes.

Reply:
Yes, that is a good idea! I will do that.